### ISSUE TITLE
Replace call to `xprop` with pure-JS code

### ISSUE THREAD

**Issue by DemiMarie** [mogh99/qubes-gnome-window-coloring#1]
The call to `xprop` is a cute trick, but it isn’t fail-safe and blocking the event loop is a bad idea.  Obtaining the window properties should be possible from JavaScript.  If it isn’t, then the answer is to implement a custom GObject in C that makes it possible!

**Comment by mogh99** [comment]
I don't think it is possible to access the window properties to get the `_QUBES_VMNAME` purely using JS.

gjs (gnome js) provides the window as `metawindow` which has some pre-specified methods to access some window props like wm_class as you can see here at [mutter](https://gitlab.gnome.org/GNOME/mutter/-/blob/main/src/meta/window.h).

However, I raised a question at [gnome-discourse](https://discourse.gnome.org/t/getting-a-window-properties-using-gjs-like-wm-class/10700) to see if there is a better way to access the window props. 

Regarding GObject I am not familiar with the idea so if you could provide some links or more explanation so I could figure my way to solve the problem.
